shakenfist/logutil.py

```diff
@@ -141,8 +161,44 @@ def process(self, msg, kwargs):
                                      traceback.extract_stack()[-4].name)
             self._extra['method'] = caller
 
+        # Emit events: set hidden=True in fields to hide this log message from end users
+        # if self.extra.get('is_event', False):
+        #     self._emit_event(msg, self._extra)
+
+        # Emit log message
+        msg = '%s[%s] %s' % (setproctitle.getproctitle(), os.getpid(), msg)
+        kwargs['extra'] = self._extra
         return msg, kwargs
 
+    def _emit_event(self, msg, extra):
+        global JSONEncoderTasks
+
+        # We only record events for known object types
+        object_count = 0
+        for objtype in OBJECT_NAMES:
+            if objtype in extra:
+                object_count += 1
+
+        if object_count == 0:
+            return
+
+        # We do not use the etcd abstraction here because it logs
+        try:
+            if not JSONEncoderTasks:
+                JSONEncoderTasks = importlib.import_module(
+                    'shakenfist.baseobject').JSONEncoderTasks
+
+            encoded = json.dumps(
+                EventLogMessageTask(time.time(), msg, extra),
+                indent=4, sort_keys=True, cls=JSONEncoderTasks)
+            Etcd3Client().put(
+                '/sf/queues/eventlog/%s/%s-%s'
+                % (config.NODE_NAME, time.time(), util_random.random_id),
+                encoded, lease=None)
+        except Exception:
+            # Don't let event errors break logging
```

Comment:
Silent exceptions lead to not so silent pain. Any exceptions here need to surface somehow.

Reply:
My issue is that an exception here can cause logging to stop working entirely, which would stop you from seeing other exceptions...